content/blog/entries/science-museum-implementation/contents.lr

```diff
@@ -0,0 +1,115 @@
+                                                (1500, 1750),
+                                                (1750, 1825),
+                                                (1825, 1850),
+                                                (1850, 1875),
+                                                (1875, 1900),
+                                                (1900, 1915),
+                                                (1915, 1940),
+                                                (1940, 1965),
+                                                (1965, 1990),
+                                                (1990, 2020)
+                                            ]
+```
+
+With this we confirm the implementation method, but before writing the script we need to know what are the different license provided 
+by them and check if it matches with our licenses and version, since they haven't mentioned about their licenses and version, running a
+test script to get all the licenses and grouping them would do the work.
+
+```
++-----------------+----------+
+| license_version | count(1) |
++-----------------+----------+
+| CC-BY-NC-ND 2.0 |      210 |
+| CC-BY-NC-ND 4.0 |     2376 |
+| CC-BY-NC-SA 2.0 |        1 |
+| CC-BY-NC-SA 4.0 |    61694 |
++-----------------+----------+
+```
+
+Since the licenses and their versions are confirmed, we can start the implementation.
+
+## Implementation
+The implementation is quite simple in nature, we loop the through the ```YEAR_RANGE``` and get all the records for that period and 
+pass it on to an object data handler method that extracts the necessary details from the record and store it in the ```ImageStore```
+instance. Since we are querying with the objects period, the same object would be present in multiple year range ,to avoid repition
+we keep track of the record/object's id in a global variable ```RECORD_IDS = []```.
+
+Within the object data handler method before collecting details we check if the ```id``` already exists in ```RECORD_IDS```, if it exists 
+we move on to the next record.
+
+```
+                                            for obj_ in batch_data:
+                                                id_ = obj_.get("id")
+                                                if id_ in RECORD_IDS:
+                                                    continue
+                                                RECORD_IDS.append(id_)
+```
+
+```id_``` is the object id and we cannnot use this value as foreign identifier, the reason behind it is that an object could
+have multiple images with it and using object id we cannot determine the image uniquely, so we must use image id that unique
+for each image. Currently image id is taken from ```multimedia```, for each image data in multimedia foreign id is in ```admin.uid```.
+
+Once the implementation script is ready (test suite is built along with), we create a workflow, that runs on a monthly basis.
```

Comment:
I suggest skipping the workflow concept, since you'd need to devote quite a few words to introducing it.